Bowling/Bowling/Frames.cs

```diff
@@ -0,0 +1,177 @@
+﻿using NUnit.Framework.Constraints;
+using NUnit.Framework.Internal.Execution;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace BowlingApp
+{
+    public class Frames
+    {
+        public const int frameNumberTotal = 10;
+        public const int frameNumberWithBonusRoll = 10;
+        private int placeCount = 0;
+        private string score = string.Empty;
+
+        public List<Frame> CreateAllFrames(string bowlingMarks)
+        {
+            List<Frame> framesCollection = new List<Frame>();
+            var marksLength = bowlingMarks.Length;
+
+            try
+            {
+                for (var frameNumber = 1; frameNumber <= frameNumberTotal; frameNumber++)
+                {
+                    if ((placeCount <= marksLength) && (frameNumber != frameNumberWithBonusRoll))
+                    {
+                        framesCollection = CreateFrameNoBonusRoll(frameNumber, bowlingMarks, framesCollection);
+                    }
+                    else
+                    {
+                        framesCollection = CreateFrameWithBonusRoll(frameNumber, bowlingMarks, marksLength, framesCollection);
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"There was a problem creating the Frames collection: {ex.Message}");
+            }
+
+            return framesCollection;
+        }
+
+        private List<Frame> CreateFrameNoBonusRoll(int frameNumber, string bowlingMarks, List<Frame> framesCollection)
+        {
+            Frame frame  = new Frame();
+            Rolls scoreRolls = new Rolls();
+
+            frame.frameNumber = frameNumber;
+            score = bowlingMarks.Substring(placeCount, 1);
+
+            frame = scoreRolls.ScoreFirstRoll(frame, score);
+            placeCount++;
+
+            if (!score.Equals(RollTypes.StrikeMark))
+            {
+                score = bowlingMarks.Substring(placeCount, 1);
+                frame = scoreRolls.ScoreSecondRoll(frame, score);
```

Comment:
This line can present some issues.  The issue is that it takes in `frame` and returns `frame` which overwrites `frame` on line 54 can lead to some interesting entanglement and unit testing trickery.  It's a bit hard to explain, but the advice I'd give here is you likely don't want to take a half-create `Frame` object as a parameter and finishing creating it in a function to then override the original `Frame` object in the parent function.  It's logically tricky as a human to understand why this is happening.  I think once you get to writing more unit tests the reason why this is a problem will become more evident.